### PR TITLE
Polling removed from endpoints on canonical

### LINKED ISSUE #7600: Change av embed origin to speed up SMP response on canonical
**Is your feature request related to a problem? Please describe.**

- Currently, due to an AMP requirement, we use the following AV embed origins:
https://polling.test.bbc.co.uk/ for test and local env
https://polling.bbc.co.uk/ for live env

- It is estimated by SMP that using `polling.bbc.co.uk` instead of the naked domain adds an extra second for 1 in 10 users. 
Jim Ley shared a sumo query exposing two areas of time saving, where removing `polling` would mean both the DNS lookup and secureconnection would not need to happen. Example given is for live radio:

Screenshot 2020-08-24 at 12 44 58

- We still need to serve the SMP iframe on an origin other than www for AMP to work, so this change should only be made for canonical

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Testing notes**

- [ ] This feature is expected to need manual testing.

### PR DESCRIPTION
**Pull request by RichardPK**
Resolves #7600

**Overall change:**
- Update `getEmbedUrl` logic to return naked `bbc` domain on canonical and `polling.bbc` on amp
- Update range of tests to reflect this change across environments
- Update CSP headers & tests for canonical changes
- Hunners of snapshots.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- This could be tested locally, checking that the polling endpoints being used to construct the av-embed outer iframe, i.e. for https://www.bbc.com/pidgin/tori-53859421, the endpoint is: https://polling.bbc.co.uk/ws/av-embeds/cps/pidgin/tori-53859421/p08p79h6/pcm.
Screenshot 2020-08-26 at 16 07 37

- These endpoints are used across: **Live Radio, On Demand Radio, On Demand TV, MAPs, Articles** (with media)
- On canonical, the endpoint should start `bbc.com/ws/av-embeds...` or `test.bbc.com/ws/av-embeds...` (according to env)
- On amp, the endpoint should start `polling.bbc.co.uk/ws/av-embeds...` or `test.polling.bbc.co.uk/ws/av-embeds...`
- In both cases - these URLs should return a valid player (as seen when visiting https://polling.bbc.co.uk/ws/av-embeds/cps/pidgin/tori-53859421/p08p79h6/pcm)

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/getEmbedUrl/index.js [hunk 1]
```diff
@@ -20,12 +22,19 @@ const shouldOverrideMorphEnv = (queryString, type) => {
   return isMediaType;
 };
 
+const getBaseUrl = isAmp => {
+  if (isLive() && isAmp) return LIVE_AMP_URL;
+  if (isAmp) return TEST_AMP_URL;
+  if (isLive()) return LIVE_CANONICAL_URL;
+  return TEST_CANONICAL_URL;
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getEmbedUrl/index.js]: How about:

```suggestion
 if (isAmp) {
   return isLive() ? LIVE_AMP_URL : TEST_AMP_URL;
 }
  return isLive() ? LIVE_CANONICAL_URL : TEST_CANONICAL_URL;
```

Or something along those lines. Or there could be another way, using a MAP / Array / Object to store the values.. Mmm 🤔

File: src/app/lib/utilities/getEmbedUrl/index.js [hunk 2]
```diff
@@ -2,8 +2,10 @@ import isLive from '../isLive';
 
 const AV_ROUTE = 'ws/av-embeds';
 
-const LIVE_URL = 'https://polling.bbc.co.uk';
-const TEST_URL = 'https://polling.test.bbc.co.uk';
+const LIVE_CANONICAL_URL = 'https://bbc.co.uk';
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getEmbedUrl/index.js]: Worth trying .com instead of .co.uk? - or actually maybe even www.bbc.com / www.test.bbc.com

**Review by ryanmccombe — APPROVED** [review]

File: src/app/lib/utilities/getEmbedUrl/index.js [hunk 3]
```diff
@@ -2,8 +2,10 @@ import isLive from '../isLive';
...
+const LIVE_AMP_URL = 'https://polling.bbc.co.uk';
+const TEST_CANONICAL_URL = 'https://test.bbc.co.uk';
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getEmbedUrl/index.js]: Worth trying .com instead of .co.uk?

File: src/app/lib/utilities/getEmbedUrl/index.js (hunk 1 above)
Comment (RichardPK) [reply src/app/lib/utilities/getEmbedUrl/index.js]: Yea nice. I knew this conditional was naaarsty, was gonna revisit - I like this suggestion

File: src/app/lib/utilities/getEmbedUrl/index.js (hunk 2 above)
Comment (RichardPK) [reply src/app/lib/utilities/getEmbedUrl/index.js]: Sorry I just _had_ to use that image. 

.com makes a lot more sense - I was just mimicking what we already had. Do you know why we had .co.uk here previously?

Comment (karinathomasbbc) [reply src/app/lib/utilities/getEmbedUrl/index.js]: It was a request from James, when I did this first time round. Prevents an unnecessary redirect?

File: src/app/lib/utilities/getEmbedUrl/index.js (hunk 3 above)
Comment (RichardPK) [reply src/app/lib/utilities/getEmbedUrl/index.js]: Let's keep chat on this to the y tho thread I just wanna see that face every time I look at this PR

**Comment by karinathomasbbc** [comment]
I also noticed that not all of these URLs work properly - get various SMP errors.

File: src/app/lib/utilities/getEmbedUrl/index.js (hunk 2 above)
Comment (karinathomasbbc) [reply src/app/lib/utilities/getEmbedUrl/index.js]: @jamesdonoh do you recall https://github.com/bbc/simorgh/pull/5493#discussion_r381363721?

**Comment by RichardPK** [comment]
> I also noticed that not all of these URLs work properly - get various SMP errors.

Ah nice can you share @karinathomasbbc ? I've been trying to run the e2es locally to validate but there's a lot of ground to cover

File: src/app/lib/utilities/getEmbedUrl/index.js (hunk 2 above)
Comment (ryanmccombe) [reply src/app/lib/utilities/getEmbedUrl/index.js]: We should use .com - WS users will be on bbc.com - if we use a .co.uk domain for the embed endpoint, that will be another DNS lookup - something this PR was intended to solve

It also seems to be the case that .co.uk redirects to .com anyway on the /ws subpath: https://www.test.bbc.co.uk/ws/av-embeds/legacy/persian/iran/2016/09/160907_tc2_testmap1/123/ru

**Comment by karinathomasbbc** [comment]
> > I also noticed that not all of these URLs work properly - get various SMP errors.
> 
> Ah nice can you share @karinathomasbbc ? I've been trying to run the e2es locally to validate but there's a lot of ground to cover

We're actually good - I checked all of the URLs in the updated snapshots and see same behaviour before and after 👍

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/lib/utilities/getEmbedUrl/index.js (hunk 2 above)
Comment (RichardPK) [reply src/app/lib/utilities/getEmbedUrl/index.js]: Hey just to clarify for future code explorers - the `.com` change only works on canonical (i.e. without the `polling.` domain). 

For AMP, with `polling` - we still need to use .co.uk

**Comment by paruchurisilpa** [comment]
Checked using these URL's:

* http://localhost:7080/pidgin/23248703
* The endpoint is  https://polling.test.bbc.co.uk/ws/av-embeds/cps/pidgin/23248703/p01kx42v/pcm which seems to have the player.

LGTM..

**Pull request merged**